dex/src/test/scala/com/wavesplatform/dex/api/AggregatedOrderBookActorSpec.scala

```diff
@@ -0,0 +1,379 @@
+    flexibleSidesOrdersGen(maxLevelsInOrderBook, maxOrdersInLevel, askPricesGen, bidPricesGen).map(Function.tupled(mkOrderBook))
+
+  private val askGen = orderGen(pricesGen, OrderType.SELL)
+  private val bidGen = orderGen(pricesGen, OrderType.BUY)
+
+  private def ordersGen(maxOrdersNumber: Int): Gen[List[AcceptedOrder]] =
+    for {
+      orderSides <- Gen.resize(maxOrdersNumber, Gen.listOf(orderSideGen))
+      orders <- Gen.sequence {
+        orderSides.map { side =>
+          val orderGen = if (side == OrderType.SELL) askGen else bidGen
+          Gen.oneOf(limitOrderGen(orderGen), marketOrderGen(orderGen))
+        }
+      }
+    } yield orders.asScala.toList
+
+  "AggregatedOrderBookActor" - {
+    "properties" - {
+      "aggregate(updatedOrderBook) == updatedAggregatedOrderBook" in forAll(orderBookGen, ordersGen(10)) { (initOb, orders) =>
+        val obsdb = OrderBookSnapshotDB.inMem
+        obsdb.update(pair, 0, Some(initOb.snapshot))
+
+        val owner = TestProbe()
+        val orderBookRef = system.actorOf(
+          Props(new OrderBookActor(
+            OrderBookActor.Settings(AggregatedOrderBookActor.Settings(100.millis)),
+            owner.ref,
+            TestProbe().ref,
+            system.actorOf(OrderBookSnapshotStoreActor.props(obsdb)),
+            pair,
+            8,
+            8,
+            time,
+            NonEmptyList.one(DenormalizedMatchingRule(0L, DenormalizedMatchingRule.DefaultTickSize)),
+            _ => (),
+            raw => MatchingRule(raw.startOffset, (raw.tickSize * BigDecimal(10).pow(8)).toLongExact),
+            _ => (t, m) => m.matcherFee -> t.matcherFee
+          )))
+        owner.expectMsgType[OrderBookActor.OrderBookRecovered]
+
+        orders.foreach(orderBookRef ! _)
+
+        orderBookRef ! MatcherActor.SaveSnapshot(100L)
+        owner.expectMsgType[OrderBookActor.OrderBookSnapshotUpdateCompleted]
+
+        val orderBookAskAdapter = new OrderBookAskAdapter(new AtomicReference(Map(pair -> Right(orderBookRef))), 5.seconds)
+
+        val updatedOrderBook = eventually {
+          val ob = obsdb.get(pair)
+          ob shouldNot be(empty)
+          ob.get._2
+        }
+
+        val actual = Await.result(orderBookAskAdapter.getAggregatedSnapshot(pair), 5.seconds).explicitGet().get
+
+        val expected = OrderBookAggregatedSnapshot(
+          asks = sum(updatedOrderBook.asks),
+          bids = sum(updatedOrderBook.bids)
+        )
+
+        actual should matchTo(expected)
+      }
+    }
+
+    "apply" - {
+      "should init with provided order book" in forAll(orderBookGen) { orderBook =>
+        val ref      = mk(orderBook)
+        val actual   = orderBookFrom { get[HttpResponse](ref)(Query.GetHttpView(MatcherModel.Normalized, 10, _)) }
+        val expected = orderBookResultFrom(orderBook)
+        actual.copy(timestamp = 0L) should matchTo(expected)
+      }
+    }
+
+    "should return update an updated" - {
+      "market status after update" - {
```

Comment:
should return _update_ an _updated_ market status after _update_
One word is superfluous

Reply:
😄👍 This test was moved three times. Will fix this